### ISSUE TITLE
Docker registry auth no longer working in 0.9 beta

### ISSUE THREAD

**Issue by ed-alertedh** [hashicorp/nomad#5412]
### Nomad version
Nomad v0.9.0-beta3 (2f55f78b21a5e55ab122f2c1e1ed1ec21fde9566)

### Operating system and Environment details
Ubuntu 16.04.5 LTS (amd64)
Docker version 18.09.3, build 774a1f4

### Issue
We are testing the latest 0.9 beta as we are keen to start using the new GPU functionality. I have found that basic authentication has stopped working for us with private docker hub images and am unsure if this has been caused by the new nomad version or by a new docker version (or something else).

This was all working with nomad 0.8.x so any help identifying the problem would be much appreciated!

### Reproduction steps
Run a job using the docker driver and a private image with basic auth. We are using the `dns_servers` option so that the container can resolve addresses using consul on the host - I don't believe this should impact the docker driver resolving `docker.io`.

### Nomad Client logs (if appropriate)
```
Failed to pull `docker.io/<dockerhub_org>/keycloak:<image tag>`: API error (404): pull access denied for <dockerhub_org>/keycloak, repository does not exist or may require 'docker login'
```

### Job file (if appropriate)
```
task "keycloak" {
    driver = "docker"

    config {
        force_pull = true
        dns_servers = [ "172.17.0.1" ]
        image = "docker.io/<dockerhub_org>/keycloak:<image tag>"
        auth {
            username = "hardcoded_username"
            password = "hardcoded_password"
        }
    }
}
```
Removing `docker.io/` from the image produces the same behaviour.

**Comment by ed-alertedh** [comment]
I have isolated the issue to this line: https://github.com/hashicorp/nomad/blob/v0.9.0-beta3/drivers/docker/utils.go#L85

The workaround for now is to add an email to the auth stanza. However, there was previously no requirement to set this field and I am unconvinced it's necessary. In addition, there is no warning or error raised and the code just silently drops the auth config.

Happy to put together a PR if we can clarify what the ideal behaviour is here.

**Comment by preetapan** [comment]
@ed-alertedh Thanks for the report, fix is now merged and will be available in 0.9.0-rc

**Comment by ed-alertedh** [comment]
@preetapan great! thanks for fixing it so quickly. Is there any release schedule at the moment or is it just "when it's ready"?
